src/Diagrams/Swimunit/Dotmatrix.hs

```diff
@@ -0,0 +1,843 @@
+               , ('6'     , dotchar_6__B6x9)
+               , ('7'     , dotchar_7__B6x9)
+               , ('8'     , dotchar_8__B6x9)
+               , ('9'     , dotchar_9__B6x9)
+               , ('A'     , dotchar_A__B6x9)
+               , ('a'     , dotchar_A__B6x9)
+               , ('B'     , dotchar_B__B6x9)
+               , ('b'     , dotchar_B__B6x9)
+               , ('C'     , dotchar_C__B6x9)
+               , ('c'     , dotchar_C__B6x9)
+               , ('D'     , dotchar_D__B6x9)
+               , ('d'     , dotchar_D__B6x9)
+               , ('E'     , dotchar_E__B6x9)
+               , ('e'     , dotchar_E__B6x9)
+               , ('F'     , dotchar_F__B6x9)
+               , ('f'     , dotchar_F__B6x9)
+               , ('G'     , dotchar_G__B6x9)
+               , ('g'     , dotchar_G__B6x9)
+               , ('H'     , dotchar_H__B6x9)
+               , ('h'     , dotchar_H__B6x9)
+               , ('I'     , dotchar_I__B6x9)
+               , ('i'     , dotchar_I__B6x9)
+               , ('J'     , dotchar_J__B6x9)
+               , ('j'     , dotchar_J__B6x9)
+               , ('K'     , dotchar_K__B6x9)
+               , ('k'     , dotchar_K__B6x9)
+               , ('L'     , dotchar_L__B6x9)
+               , ('l'     , dotchar_L__B6x9)
+               , ('M'     , dotchar_M__B6x9)
+               , ('m'     , dotchar_M__B6x9)
+               , ('N'     , dotchar_N__B6x9)
+               , ('n'     , dotchar_N__B6x9)
+               , ('O'     , dotchar_O__B6x9)
+               , ('o'     , dotchar_O__B6x9)
+               , ('P'     , dotchar_P__B6x9)
+               , ('p'     , dotchar_P__B6x9)
+               , ('Q'     , dotchar_Q__B6x9)
+               , ('q'     , dotchar_Q__B6x9)
+               , ('R'     , dotchar_R__B6x9)
+               , ('r'     , dotchar_R__B6x9)
+               , ('S'     , dotchar_S__B6x9)
+               , ('s'     , dotchar_S__B6x9)
+               , ('T'     , dotchar_T__B6x9)
+               , ('t'     , dotchar_T__B6x9)
+               , ('U'     , dotchar_U__B6x9)
+               , ('u'     , dotchar_U__B6x9)
+               , ('V'     , dotchar_V__B6x9)
+               , ('v'     , dotchar_V__B6x9)
+               , ('W'     , dotchar_W__B6x9)
+               , ('w'     , dotchar_W__B6x9)
+               , ('X'     , dotchar_X__B6x9)
+               , ('x'     , dotchar_X__B6x9)
+               , ('Y'     , dotchar_Y__B6x9)
+               , ('y'     , dotchar_Y__B6x9)
+               , ('Z'     , dotchar_Z__B6x9)
+               , ('z'     , dotchar_Z__B6x9)
+               , ('\0'    , dotchar_Unmapped_B6x9) -- Only mapping that unconditionally has to be present.
+               ])
+
+dotchar_Hash_B6x9 :: Dotchar
+dotchar_Hash_B6x9 = Dotchar
+              ["      "
+              ," # #  "
+              ," # #  "
+              ,"##### "
+              ," # #  "
+              ,"##### "
+              ," # #  "
+              ," # #  "
+              ,"      "]
```

Comment:
You could store each character as a 'Word64' where the first 54 bits correspond to whether a dot is on or off, so making a font doesn't take so many lines. Just a suggestion.